wandb/sdk/wandb_init.py

```diff
@@ -345,44 +345,74 @@ def getcaller():
+    allow_val_change: bool = None,  # LB: What is this?
+    resume: Union[bool, str, None] = None,
+    force: bool = None,
+    tensorboard: bool = None,  # alias for sync_tensorboard, deprecated
+    sync_tensorboard: bool = None,
+    monitor_gym: bool = None,  # LB: Currently unsupported
+    id: Optional[str] = None,
+    settings: Union[Settings, Dict[str, Any], str, None] = None,  # LB: What is this?
+) -> Run:  # LB: save_code seems unsupported?
+    """Initialize W&B
+
+    Spawns a new process to communicate with W&B.
+
+    If called from within Jupyter, initializes a new run and waits for a call to
+    `wandb.log` to begin pushing metrics.
 
     Args:
-        entity: alias for team.
-        team: personal user or team to use for Run.
-        project: project name for the Run.
-
-    Raises:
-        Exception: if problem.
+        job_type (str, optional): The type of job running, defaults to 'train'
+        dir (str, optional): An absolute path to a directory where metadata will be
+            stored
+        config (dict, argparse, or tf.FLAGS, optional): The hyper parameters to
+            store with the run
+        project (str, optional): The run's project
+        entity (str, optional): The run's entity
+        reinit (bool, optional): Allow multiple calls to init in the same process
+        tags (list, optional): A list of tags to apply to the run
+        team (str, optional): ??
+        group (str, optional): A unique string shared by all runs in a given group
+        name (str, optional): A display name which does not have to be unique.
+        notes (str, optional): A multiline string associated with the run
+        magic (bool, dict, or str, optional): magic configuration as bool, dict,
+            json string, yaml filename
+        config_exclude_keys (list, optional): string keys to exclude storing in W&B
+            when specifying config.
+        config_include_keys (list, optional): string keys to include storing in W&B
+            when specifying config
+        anonymous (str, optional): Can be "allow", "must", or "never". Controls whether
+            anonymous logging is allowed.  Defaults to never.
+        resume (bool, str, optional): Automatically resume this run if run from the
+            same machine, you can also pass a unique run_id
+        force (bool, optional): Force authentication with wandb, defaults to False
+        sync_tensorboard (bool, optional): Synchronize wandb logs to tensorboard or
+            tensorboardX
+        save_code (bool, optional): Save the entrypoint or jupyter session history
+            source code.
+        monitor_gym (bool, optional): automatically log videos with openai gym
+            (https://gym.openai.com/)
+        id (str, optional): A globally unique (per project) identifier for the run
+        settings (dict): ??
 
     Returns:
-        wandb Run object
-
+        A wandb.run object for metric and config logging.
```

Comment:
This is actually a `wandb.wandb_sdk.wandb_run.ManagedRun` object.  We might want to expose this in the global namespace as `Run` so users could do `help wandb.Run`.  Another gotcha is that sometimes this is a DummyRun if we're in noop mode.  @raubitsj thoughts?

Reply:
ManagedRun and DummyRun will be unified as Run... coming soon